code_sync/code_sync.py

```diff
@@ -2,39 +2,186 @@
 
 import argparse
 import os
+from pathlib import Path
 import subprocess
+import yaml
+from typing import Dict, Union
+
 
 cmd_str = 'watchmedo shell-command --recursive --patterns="{local_dir}*" --command="rsync --filter=\':- .gitignore\' ' \
           '--exclude \'*.ipynb\' --exclude \'.git\' --delete-after -rz --port {port} {local_dir} ' \
           '{target}:{remote_dir}" {local_dir}'
 
 epilog_str = '''
-Example for connecting to LeoMed:
-    code_sync --local_dir mylocaldir/ --remote_dir myremotedir/ --target medinfmk --port 2222\n
+EXAMPLE USAGE
+Register a project:
+    code_sync --register <project>
+
+code_sync a registered project:
+    code_sync <project>
+
+List all projects registered to code_sync:
+    code_sync --list
+
+Run code_sync with specific parameters:
+    code_sync --local_dir <mylocaldir/> --remote_dir <myremotedir/> --target <ssh_remote> --port 2222\n
 
 '''
 
+CONFIG_FILE_NAME = '.code_sync'
+
 
 def code_sync(local_dir, remote_dir, target, port=22):
     # clean up slashes
     local_dir = os.path.join(local_dir, '')
     remote_dir = os.path.join(remote_dir, '')
 
-    # subprocess.call()
+    print("Starting code_sync between {} and {}:{} ...".format(local_dir, target, remote_dir))
+    print('(^C to quit)')
     cmd = cmd_str.format(local_dir=local_dir, remote_dir=remote_dir, target=target, port=port)
     subprocess.call(cmd, shell=True)
 
 
+def load_config() -> Dict:
+    """
+    Load the code_sync config file. Create a blank one if no file exists.
+
+    Returns:
+        The config loaded from the file.
+    """
+
+    create_config_if_not_exists()
+
+    config_file_path = Path(Path.home(), CONFIG_FILE_NAME)
```

Comment:
The path.home is it the pwd of the command line?

Reply:
no, it's the user's home directory, `$HOME`

Reply:
So there is only one .code_sync file per user?
I though the .code_sync file would be something living inside the repository, so that it can be shared across people in the lab... so each git repo gets it's own .code_sync file

Reply:
Yes, different users will have their git repositories at their own locations on their file system. For example, with MRIdle- my MRIdle repo is at `~/Documents/githubs/mridle/`, but Nico sometimes works from a Windows machine, so his path for his MRIdle repo is completely different. I don't think we want to enforce that everyone in the project has to have their paths be the same. 

For the functionality that you are working on- that automates the singularity image etc- that may want to be shared, but I think that should be separate from code_sync.

Reply:
Ok, everything clear tehn